### ISSUE TITLE
Performance: improve performance of real-time events

### ISSUE THREAD

**Issue by metalelf0** [extr0py/oni#174]
The new update (0.1.8) significantly improved performances, making oni really usable as a daily driver, but some "interface heavy" events (like window resizing, or search-as-you-type in FZF.vim) still feel really slow compared to terminal neovim and neovim-qt. Thanks!

**Comment by extr0py** [comment]
Yeah, totally, we need to fix some of those performance issues still. Thanks for the report!

There are a few potential bottlenecks I saw in my last round of profiling:
1) Need to add incremental render checks (sometimes we update the elements too aggressively, even if they haven't changed)
2) Scrolling needs to be optimized via #88 
3) In some cases, the it looks like the traversal of data we get via msgpack-rpc can be the bottleneck - need to investigate that further

**Comment by extr0py** [comment]
Integrating the `LiveUpdate` will significantly help performance: https://github.com/neovim/neovim/pull/5269

A major drain on performance is the processing of our `msgpack` messages, and in the case of buffer updates, we are sending the entire buffer in most normal-mode cases.

**Comment by benjamin-rood** [comment]
I'm getting some (tolerable) keyboard input lag on very big files from this issue in Oni, and monstrously large input lag in VS Code using the VSCodeVim extension. Hopefully the `LiveUpdate` PR will help with that too.

**Comment by extr0py** [comment]
@benjamin-rood - thanks for letting us know! Yes, the `LiveUpdate` should offer big performance improvements for us... especially in cases when we are sending the whole buffer over (like, any normal mode operation).

Just a couple of questions:
- Are you seeing input lag for particular gestures? (Typing in insert mode, moving in normal mode, scrolling?)
- What OS are you on?
- How big is the file?

In addition, are you seeing any lag on small files?

Want to make sure I'm capturing everything in my profiling.
